### ISSUE TITLE
[a11y] - Result List -

### ISSUE THREAD

**Issue by transkamp** [open-pioneer/trails-openlayers-base-packages#443]
### Issue description

https://open-pioneer.github.io/trails-demo/openlayers-base-packages/dev/samples/showcase/?demo=selectionResultList

==> missing `role="region"` `aria-label="Result List"` for result list section. Improves screenreader navigation
https://www.w3.org/WAI/ARIA/apg/patterns/landmarks/examples/HTML5.html

################

An aria-labelledby attribute doesn't match any element id. This might prevent the browser from correctly autofilling the form and accessibility tools from working correctly.
![Image](https://github.com/user-attachments/assets/87c29304-e78a-4f46-acb3-500628fa753c)

###############

checkbox needs more context. it is easy to understand if the checkbox contains further information about which element is selected. this can be realized if the content of one or more cells is joined with the checkbox. of course, this would then have to be solved by configuration. 
![Image](https://github.com/user-attachments/assets/0bb794e3-2396-4419-9d20-d1389dd98e16)

##############

If a result list can only be created by dragging a BBox on the map with a mouse, then it is necessary to inform both sighted and blind users of this fact.
![Image](https://github.com/user-attachments/assets/1fe499e8-497d-466c-bf03-e2c5b0bc7495)

### Validations

- [x] Read and follow our [code of conduct](https://github.com/open-pioneer/.github/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [contributing guidelines](https://github.com/open-pioneer/trails-starter/blob/main/docs/Contributing.md)
- [x] Check that there isn't already a similar issue
